### PR TITLE
Generate SBOM for static binary bundle

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now create a bill of materials for the static binary bundle which gets uploaded to our bucket. The verification of the sbom will be part of another PR.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
We now provide a bill of materials for the static binary bundle.
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
cc @puerco

**Comment by saschagrunert** [comment]
@puerco is this run normal, especially the warnings concern me:

https://github.com/cri-o/cri-o/runs/7003855581?check_suite_focus=true

```
…
level=warning msg="Unknown tag: FileType"
level=warning msg="Unknown tag: FileType"
level=warning msg="Unknown tag: FileType"
level=warning msg="Unknown tag: FileType"
level=warning msg="Unknown tag: FileType"
level=warning msg="Unknown tag: FileType"
level=warning msg="Unknown tag: FileType"
level=warning msg="Unknown tag: FileType"
level=warning msg="Unknown tag: FileType"
level=warning msg="Unknown tag: FileType"
level=warning msg="Unknown tag: FileType"
++ install -d -m 755 test//etc/cni/net.d
/ __| '_ \ / _` \ \/ /
\__ \ |_) | (_| |>  < 
|___/ .__/ \__,_/_/\_\
    |_|               
 📂 SPDX Document CRI-O
  │ 
  │ 📦 DESCRIBES 1 Packages
  │ 
  ├ cri-o
  │  │ 🔗 43 Relationships
  │  ├ CONTAINS FILE README.md (README.md)
  │  ├ CONTAINS FILE Makefile (Makefile)
  │  ├ CONTAINS FILE bin/conmon (bin/conmon)
  │  ├ CONTAINS FILE bin/pinns (bin/pinns)
  │  ├ CONTAINS FILE bin/crun (bin/crun)
  │  ├ CONTAINS FILE cni-plugins/bandwidth (cni-plugins/bandwidth)
  │  ├ CONTAINS FILE bin/runc (bin/runc)
  │  ├ CONTAINS FILE cni-plugins/bridge (cni-plugins/bridge)
  │  ├ CONTAINS FILE cni-plugins/firewall (cni-plugins/firewall)
  │  ├ CONTAINS FILE cni-plugins/host-device (cni-plugins/host-device)
  │  ├ CONTAINS FILE cni-plugins/dhcp (cni-plugins/dhcp)
  │  ├ CONTAINS FILE cni-plugins/host-local (cni-plugins/host-local)
  │  ├ CONTAINS FILE cni-plugins/ipvlan (cni-plugins/ipvlan)
  │  ├ CONTAINS FILE cni-plugins/loopback (cni-plugins/loopback)
  │  ├ CONTAINS FILE cni-plugins/macvlan (cni-plugins/macvlan)
  │  ├ CONTAINS FILE cni-plugins/portmap (cni-plugins/portmap)
  │  ├ CONTAINS FILE cni-plugins/ptp (cni-plugins/ptp)
  │  ├ CONTAINS FILE cni-plugins/sbr (cni-plugins/sbr)
  │  ├ CONTAINS FILE cni-plugins/static (cni-plugins/static)
  │  ├ CONTAINS FILE cni-plugins/tuning (cni-plugins/tuning)
  │  ├ CONTAINS FILE bin/crictl (bin/crictl)
  │  ├ CONTAINS FILE completions/bash/crio (completions/bash/crio)
  │  ├ CONTAINS FILE completions/bash/crio-status (completions/bash/crio-status)
  │  ├ CONTAINS FILE completions/fish/crio-status.fish (completions/fish/crio-status.fish)
  │  ├ CONTAINS FILE completions/fish/crio.fish (completions/fish/crio.fish)
  │  ├ CONTAINS FILE completions/zsh/_crio (completions/zsh/_crio)
  │  ├ CONTAINS FILE completions/zsh/_crio-status (completions/zsh/_crio-status)
  │  ├ CONTAINS FILE contrib/10-crio-bridge.conf (contrib/10-crio-bridge.conf)
  │  ├ CONTAINS FILE contrib/crio.service (contrib/crio.service)
  │  ├ CONTAINS FILE contrib/policy.json (contrib/policy.json)
  │  ├ CONTAINS FILE etc/10-crun.conf (etc/10-crun.conf)
  │  ├ CONTAINS FILE etc/crictl.yaml (etc/crictl.yaml)
  │  ├ CONTAINS FILE etc/crio-umount.conf (etc/crio-umount.conf)
  │  ├ CONTAINS FILE etc/crio.conf (etc/crio.conf)
  │  ├ CONTAINS FILE install (install)
  │  ├ CONTAINS FILE man/crio-status.8 (man/crio-status.8)
  │  ├ CONTAINS FILE man/crio.8 (man/crio.8)
  │  ├ CONTAINS FILE man/crio.conf.5 (man/crio.conf.5)
  │  ├ CONTAINS FILE man/crio.conf.d.5 (man/crio.conf.d.5)
  │  ├ CONTAINS FILE cni-plugins/vrf (cni-plugins/vrf)
  │  ├ CONTAINS FILE cni-plugins/vlan (cni-plugins/vlan)
  │  ├ CONTAINS FILE bin/crio-status (bin/crio-status)
  │  └ CONTAINS FILE bin/crio (bin/crio)
  │ 
  └ 📄 DESCRIBES 0 Files
…
```

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -159,6 +159,14 @@ jobs:
       - build-static-arm64
     steps:
       - uses: actions/checkout@v3
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            /tmp/spdx
```
Comment (haircommander) [inline .github/workflows/test.yml:167]: where does this path come from?

Comment (saschagrunert) [reply .github/workflows/test.yml:167]: This is hardcoded into the bom tool: 

https://github.com/kubernetes-sigs/bom/blob/436b95e99c9348a8c3ee459cda81daff50f1b28f/pkg/spdx/spdx.go#L93

Comment (haircommander) [reply .github/workflows/test.yml:167]: would you add a comment ?

Comment (saschagrunert) [reply .github/workflows/test.yml]: Done :+1:

**Comment by saschagrunert** [comment]
@puerco confirmed via slack that the warning messages should be more like debug ones, so I guess we're all set. 

PTAL @haircommander

**Pull request merged**